Core/Net/NetFileCache.cs

```diff
@@ -11,43 +11,54 @@
 
 namespace CKAN
 {
-
     /// <summary>
     /// A local cache dedicated to storing and retrieving files based upon their
     /// URL.
     /// </summary>
 
     // We require fancy permissions to use the FileSystemWatcher
-    [PermissionSet(SecurityAction.Demand, Name="FullTrust")]
+    [PermissionSet(SecurityAction.Demand, Name = "FullTrust")]
     public class NetFileCache : IDisposable
     {
         private FileSystemWatcher watcher;
         private string[] cachedFiles;
         private string cachePath;
         private static readonly TxFileManager tx_file = new TxFileManager();
-        private static readonly ILog log = LogManager.GetLogger(typeof (NetFileCache));
+        private static readonly ILog log = LogManager.GetLogger(typeof(NetFileCache));
 
-        public NetFileCache(string _cachePath)
+        public NetFileCache(string _cachePath = null)
```

Comment:
This parameter is still being passed from KSP.cs:61, so this change would not yet have an effect in the full app. The `KSP.DownloadCacheDir()` function probably needs to be retired, with some kind of migration process for old downloads.

Reply:
Rewrote the `KSP.DownloadCacheDir()` and cache gets moved when new path was specified.